src/commonMain/kotlin/au/id/micolous/metrodroid/util/StreamUtils.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * StreamUtils.kt
+ *
+ * Copyright (C) 2019 Google
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package au.id.micolous.metrodroid.util
+
+import kotlinx.io.ByteArrayOutputStream
+import kotlinx.io.InputStream
+import kotlinx.io.charsets.Charsets
+
+fun InputStream.fullRead() : ByteArray {
+    val bo = ByteArrayOutputStream()
```

Comment:
`ByteArrayOutputStream` effectively removes the file size cap (64 KiB) we had before.  I understand this is because Farebot JSON dump files can contain multiple uncompressed card dumps.

If someone loads a large file that starts with JSON characters, this will make the app crash and they won't know why.  I've seen plenty of Play bug reports where someone has tried to load a large file.

Can we have a "reasonable" cap on file size here, say 4 MiB?

At the moment, Kotlin-on-Java will try to allocate up to 2 GiB, and `java.io.ByteArrayOutputStream.grow` causes us to repeatedly double our buffer size during writes -- and few devices will have 4 GiB _available_ for us to use.

It would also be a good idea to prime this buffer to be the correct size for the data we're planning to put in it, to avoid repeated `.grow` calls (and thus a copy of all the data on every expansion).

Reply:
Added a warning/confirmation.